src/miner.cpp

```diff
@@ -143,7 +143,31 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const CScript& s
       nBlockSize += h.GetSerializeSize(SER_NETWORK, PROTOCOL_VERSION);
     }
     assert(nBlockSize == 80);  // BU always 80 bytes
-    nBlockSize += MAX_COINBASE_SCRIPTSIG_SIZE;  // BU Miners may take the block we give them and add additional strings to the coinbase
+
+
+    unsigned int nCoinbaseSize=0;
+    // Compute coinbase transaction WITHOUT FEES just to get its size.  We will recompute this at the end when we know the fees.
+    {
+      txNew.vout[0].nValue = 0;  // Will be fixed below, but we need to adjust the size for a possible 9 byte varint
+      txNew.vin[0].scriptSig = CScript() << ((int) 0) << CScriptNum(0);  // block height will be fixed below, but we need to adjust the size for a possible 9 byte varint
+
+      // BU005 add block size settings to the coinbase
+      std::string cbmsg = FormatCoinbaseMessage(BUComments, minerComment);
+      const char* cbcstr = cbmsg.c_str();
+      vector<unsigned char> vec(cbcstr, cbcstr+cbmsg.size());
+      COINBASE_FLAGS = CScript() << vec;
```

Comment:
This isn't thread safe. Also, capitalized names by convention are reserved for macros in C/C++ so that's confusing.